RationaleMCP/0027/HindleyMilner.md

```diff
@@ -0,0 +1,246 @@
+# Unit checking
+
+Unit checking is the process of inferring the units of Modelica variables, and ensuring the variables are used consistently in expressions and equations.
+
+It takes place after any evaluation of evaluable parameters and no evaluable parameters shall be evaluated during this process.
+
+It is performed by using a type inference algorithm, albeit adapted to work with units.
+The algorithm proposed here is based on the Hindley-Milner algorithm.
+At a high level the algorithm collects a set of constraints on the units of expressions.
+It extracts a sequence of substitutions from these constraints which once applied constructs the unit of a given expression.
```

Comment:
To me, this looks like an informal overview of the process.  Shouldn't it be described in some more detail in the _Unit inference section_?  Alternatively, the _Unit inference_ could be renamed _Unit constraints_, and then we could have a new _Unit inference_ section afterwards where the processing of the constraints is described.  This would be beneficial for the structure in case we need to expand on this topic be, for example, explaining in more detail when a constraint is ready to be processed, or the general inability to process constraints one at a time without the need for substitutions involving other unit variables.